message.go

```diff
@@ -0,0 +1,180 @@
+package conventionalcommit
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const hash = 35 // ASCII hash ("#") character.
+
+var (
+	Err             = errors.New("conventionalcommit")
+	ErrEmptyMessage = fmt.Errorf("%w: empty message", Err)
+)
+
+// HeaderToken will match a Conventional Commit formatted subject line, to
+// extract type, scope, breaking change (bool), and description.
+//
+// It is intentionally VERY forgiving so as to be able to extract the various
+// parts even when things aren't quite right.
+var HeaderToken = regexp.MustCompile(
```

Comment:
I think it would be better to expose helper functions for checking if a `string`/`[]byte` matches a header/footer token, and also functions that extract the component parts in a standardized way.

Otherwise the capture groups of these regexp patterns effectively become part of the package's public API, making it difficult to change. Hence this public interface should be provided as helper functions instead, meaning the underlying implementation can be safely fixed and/or improved without breaking backwards compatibility.